src/Compilers/Core/Portable/InternalUtilities/SpecializedCollections.Empty.Set.cs

```diff
@@ -10,8 +10,11 @@ internal partial class SpecializedCollections
         private partial class Empty
         {
             internal class Set<T> : Collection<T>, ISet<T>
+#if COMPILERCORE
```

Comment:
These ifdefs are ugly but most expedient way to make progress against the feedback. I can fix it up by tweaking closed project to include IReadOnlySet.cs after this is merged.